src/pages/admin/donate/components/generic-details/GenericDetails.tsx

```diff
@@ -0,0 +1,231 @@
+import NotFoundIcon from '../../../../../assets/icons/not-found.svg';
+import { COMMON_TEXT_ADMIN } from '../../../../../const/admin/common';
+import './GenericDetails.scss';
+import { FieldValues } from 'react-hook-form';
+import { GenericFormMode, GenericFormProps, GenericFormRef } from '../generic-form/GenericForm';
+import { DONATE_TEXT } from '../../../../../const/admin/donate';
+import { InlineLoader } from '../../../../../components/common/inline-loader/InlineLoader';
+
+export interface GenericDetailsProps<T extends FieldValues> {
+    title?: string;
+    items: T[];
+    isLoading: boolean;
+    FormComponent: React.ForwardRefExoticComponent<
+        React.PropsWithoutRef<GenericFormProps<T>> & React.RefAttributes<GenericFormRef>
+    >;
+    primaryAddButton?: boolean;
+    initialIsItemsExpanded?: boolean;
+    notFoundText?: string;
+    addNewText: string;
+    createEmptyItem: (data: any) => T;
+    isChildForm?: boolean;
+    children?: (form: { formState: T; isItemsExpanded: boolean }) => React.ReactNode;
+    onChangeItems?: React.Dispatch<React.SetStateAction<T[]>>;
+    onSubmit?: (data: T) => Promise<void>;
+    onUpdate?: (id: number, data: T) => Promise<void>;
+    onDelete?: (id: number) => Promise<void>;
+}
+
+export function GenericDetails<T extends { id?: number } & FieldValues>({
+    title,
+    items,
+    isLoading,
+    FormComponent,
+    primaryAddButton,
+    initialIsItemsExpanded = true,
+    notFoundText,
+    addNewText,
+    createEmptyItem,
+    isChildForm = false,
+    children,
+    onChangeItems,
+    onSubmit,
+    onUpdate,
+    onDelete,
+}: GenericDetailsProps<T>) {
+    const addformRef = useRef<GenericFormRef>(null);
+    const [isAddFormVisible, setIsAddFormVisible] = useState(false);
+    const [isItemsExpanded, setIsItemsExpanded] = useState(initialIsItemsExpanded);
+
+    const updateItems = useCallback(
+        (newItems: T[]) => {
+            onChangeItems?.(newItems);
+        },
+        [onChangeItems],
+    );
+
+    const handleDelete = useCallback(
+        (id?: number) => {
+            updateItems(items.filter((i) => i.id !== id));
+        },
+        [items, updateItems],
+    );
+
+    const handleAdd = () => {
+        setIsAddFormVisible(true);
+    };
+
+    const handleClose = () => {
+        setIsAddFormVisible(false);
+    };
+
+    const handleSubmit = useCallback(
+        async (data: T) => {
+            if (onSubmit) {
+                await onSubmit(data);
+                setIsAddFormVisible(false);
+            } else if (isChildForm) {
+                updateItems([...items, data]);
+                setIsAddFormVisible(false);
+            } else {
+                const newItem = createEmptyItem({
+                    ...data,
+                    id: data.id || Date.now(),
+                    correspondentBanks: data.correspondentBanks || [],
+                } as any);
+
+                updateItems([...items, newItem]);
```

Comment:
Do we need `correspondentBanks` here?
```suggestion
                const newItemWithId = { ...data, id: data.id || Date.now() };
                updateItems((prevItems) => [...prevItems, newItemWithId]);
```